### PR TITLE
fix bugs

### PR DESCRIPTION
**Pull request by dvdli**
#197 
#184

### CODE REVIEW

**Comment by dvdli** [comment]
@tay10r @gkasten @simonrules Could you please help to review the patch set?

The Travis CI build always fails due to the python script. 

```
0.16s$ pip3 install --user ninja
Traceback (most recent call last):
  File "/home/travis/.local/bin/pip3", line 7, in <module>
    from pip._internal.cli.main import main
  File "/home/travis/.local/lib/python3.5/site-packages/pip/_internal/cli/main.py", line 60
    sys.stderr.write(f"ERROR: {exc}")
                                   ^
SyntaxError: invalid syntax
The command "pip3 install --user ninja" failed and exited with 1 during .
```

@tay10r Do you know why?

**Comment by dvdli** [comment]
I found there is a log in the previous passed build.
```
DEPRECATION: Python 3.5 reached the end of its life on September 13th, 2020. Please upgrade your Python as Python 3.5 is no longer maintained. pip 21.0 will drop support for Python 3.5 in January 2021. pip 21.0 will remove support for this functionality.
```

**Comment by tay10r** [comment]
@dvdli I spun up a Xenial VM and ran the Travis CI instructions, got the same error. I think maybe it's just type to upgrade the Travis CI distro to Bionic? Building without Ninja is fine but if the Meson build system is being supported, then I'm not sure if there's any other option.

File: tests/src/mixer_test.cc [hunk 1]
```diff
@@ -288,7 +288,7 @@ TEST_P(MixerControlsTest, Event) {
 
     EXPECT_EQ(mixer_wait_event(mixer_object, 1000), 1);
 
-    EXPECT_EQ(mixer_consume_event(mixer_object), 0);
+    EXPECT_EQ(mixer_consume_event(mixer_object), 1);
```
Comment (tay10r) [inline tests/src/mixer_test.cc:291]: For anyone else reading this, 1 means success, 0 means no pending event, and -errno may be returned in case of an error (according to the doc string).

File: tests/src/pcm_out_test.cc [hunk 2]
```diff
@@ -192,13 +192,13 @@ TEST_F(PcmOutMmapTest, Write) {
...
 
-    int written_frames = 0;
+    int res = 0;
     unsigned int frames = pcm_bytes_to_frames(pcm_object, buffer_size);
     pcm_start(pcm_object);
     auto start = std::chrono::steady_clock::now();
     for (uint32_t i = 0; i < write_count; ++i) {
-        written_frames = pcm_mmap_write(pcm_object, buffer.get(), buffer_size);
-        ASSERT_EQ(written_frames, frames);
+        res = pcm_mmap_write(pcm_object, buffer.get(), buffer_size);
```
Comment (tay10r) [inline tests/src/pcm_out_test.cc:200]: Why is this changed to zero?

File: utils/tinymix.c [hunk 3]
```diff
@@ -370,7 +370,7 @@ static void tinymix_set_byte_ctl(struct mixer_ctl *ctl,
 
 static int is_int(const char *value)
 {
-    return value[0] >= '0' || value[0] <= '9';
```
Comment (tay10r) [inline utils/tinymix.c:373]: Whoops! Good catch.

**Review by tay10r — APPROVED** [review]

File: tests/src/pcm_out_test.cc (hunk 2 above)
Comment (dvdli) [reply tests/src/pcm_out_test.cc:200]: That is because of backward compatibility. In the downstream projects, the callers always treat non-zero return value of pcm_mmap_read/write as error. We should keep the behavior of these old APIs and let callers want to get real written frames in return value use the pcm_readi/writei. 

Note 1: pcm_readi/writei can deal with pcm with MMAP flag.
Note 2: I also mark the pcm_mmap_read/write as deprecated functions.

**Comment by dvdli** [comment]
@tay10r Thanks! I works now. I changed dist form xenial to bionic.
